test/BulletDynamics/test_deterministic_save_and_restore.cpp

```diff
@@ -0,0 +1,221 @@
+#include "../../Extras/Serialize/BulletFileLoader/bFile.h"
+#include <btBulletDynamicsCommon.h>
+#include "../../src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.h"
+#include "../../src/BulletDynamics/Featherstone/btMultiBodyConstraintSolver.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btMultiBodyWorldImporter.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btBulletWorldImporter.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btWorldImporter.h"
+#include "../../src/BulletCollision/CollisionDispatch/btCollisionObject.h"
+#include "../../src/BulletCollision/CollisionDispatch/btCollisionWorld.h"
+#include "LinearMath/btVector3.h"
+#include "LinearMath/btAlignedObjectArray.h"
+#include "../../src/BulletDynamics/Dynamics/btRigidBody.h"
+
+#include "../../examples/CommonInterfaces/CommonRigidBodyBase.h"
+#include <gtest/gtest.h>
+#define USE_MOTIONSTATE 0
+
+static btMultiBodyDynamicsWorld *create_btMultiBodyDynamicsWorld()
+{
+	btDefaultCollisionConfiguration * collisionConfiguration = new btDefaultCollisionConfiguration();
+
+	btCollisionDispatcher * dispatcher = new btCollisionDispatcher(collisionConfiguration);
+
+	btDbvtBroadphase * broadphase = new btDbvtBroadphase();
+
+	btMultiBodyConstraintSolver* solver = new btMultiBodyConstraintSolver();
+
+	btMultiBodyDynamicsWorld * world = new btMultiBodyDynamicsWorld(dispatcher, broadphase, solver, collisionConfiguration);
+
+	world->setGravity(btVector3(0, -10, 0));
+
+	return world;
+
+}
+
+GTEST_TEST(BulletDynamics, DeterministicSaveRestore)
+{
+
+	static char filename[] = "test_serialize.mixail";
+	btRigidBody* box_before[3];
+	btCollisionObject* box_after[3];
```

Comment:
Why are before/after of different types?

Also a note, it's usually not a good practice to have numbers in code. They're called "magic numbers" because they have no explanation and therefore presumably have some magical properties. They are usually replaced with #define, like
```
#define NUMBER_OF_BOXES[3]
btRigidBody *box_before[NUMBER_OF_BOXES];
```
However in this case I don't think it needs to change, I think it's reasonably clear from the code that there's just 3 boxes. Just wanted to add the note so that you keep it in mind as you're writing code in the future.